### PR TITLE
feat: close connections that exceeds opts.idleTimeout

### LINKED ISSUE #60: Can we support pool for sqlite?
我们可以支持连接池吗？在一些老的 win7 平台，当有大量数据写入时，同时查询会变的特别慢。当我使用 sequelize 时，没有问题。是否可以使用 [node-pool](https://github.com/coopernurse/node-pool)来优化这个问题[](url)

### LINKED ISSUE #148: idleTimeout should be added to close idle connections
...to prevent connection squatting. The default max connections in mysql is 100-151:

```bash
mysql> show variables like 'max_connections';
+-----------------+-------+
| Variable_name   | Value |
+-----------------+-------+
| max_connections | 151   |
+-----------------+-------+
1 row in set (0.03 sec)
```

- https://stackoverflow.com/questions/22297773/how-to-increase-mysql-connectionsmax-connections

### PR DESCRIPTION
**Pull request by cyjake**
fixes #148, closes #60 with a mocked Pool and Connection for sqlite driver

### CODE REVIEW

File: src/drivers/mysql/index.js [hunk 1]
```diff
@@ -82,6 +95,8 @@ class MysqlDriver extends AbstractDriver {
     } catch (err) {
       logger.logQueryError(sql, err, Date.now() - start, opts);
       throw err;
+    } finally {
+      if (!opts.connection) connection.release();
```
Comment (JimmyDaddy) [inline src/drivers/mysql/index.js:99]: `this.pool.release(connection)` mysql2 pool may use current connection again

Comment (cyjake) [reply src/drivers/mysql/index.js:99]: It is intended to be re-used by default. We don't need to connect/disconnect on every single query

File: src/drivers/sqlite/index.js [hunk 2]
```diff
@@ -70,39 +70,88 @@ class Connection {
...
+    await new Promise(resolve => queue.push(resolve));
+    return await this.getConnection();
+  }
+
+  releaseConnection(connection) {
+    connection.idle = true;
+    this.emit('release', connection);
+
+    const { queue } = this;
+    while (queue.length > 0) {
```
[the remark this answers is not in the document]
Comment (cyjake) [reply src/drivers/sqlite/index.js:129]: It's a simple hack for referencing hanging awaits in `getConnection`, when new connection is available, begin dequeue continue previous blocking jobs.

File: src/drivers/sqlite/index.js [hunk 3]
```diff
@@ -70,39 +70,88 @@ class Connection {
...
+      }
+    }
+    if (connections.length < options.connectionLimit) {
+      const connection = new Connection({ ...options, client, pool: this });
+      connections.push(connection);
+      this.emit('acquire', connection);
+      return connection;
+    }
+    await new Promise(resolve => queue.push(resolve));
+    return await this.getConnection();
```
[the remark this answers is not in the document]
Comment (cyjake) [reply src/drivers/sqlite/index.js:121]: This is the default behavior that the mysql client supports, when connection pool is drained, wait for connection being released.

**Pull request merged**